### ISSUE TITLE
YOLO dataset validation behavior doesn't match the documentation when searching for a data.yml file

### ISSUE THREAD

**Issue by LucasGenoud** [roboflow/rf-detr#775]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

When using a yolo based dataset format, the documentation says that the library will look for either a data.yml or data.yaml file. However, the `is_valid_yolo_dataset` function used to validate the YOLO dataset only looks for a data.yaml file. 

### Environment

RF-DETR : 1.5.1

### Minimal Reproducible Example
```
model.train(
        dataset_dir="/path/to/yolo/dataset",
        epochs=args.epochs,
        batch_size=args.batch_size,
        grad_accum_steps=args.grad_accum_steps,
        patience=args.patience,
        lr=args.lr,
        output_dir=args.output_dir,
        dist_backend='gloo'
    )
```
### Additional

I don't know if the correct fix is to simply add a "data.yaml" to the list of accepted files or make a more generic validator that : 
- Look for the presence of at least one .yaml / .yml file. 
- Parses the file to extract classes as well as labels and images folder path

Currently, the folder name for the train and validation folder are hard coded but could be read for the train and val property in the yaml config file.

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!
